transport/tchannel/config_test.go

```diff
@@ -0,0 +1,242 @@
+		cfg  attrs             // outbounds section of the config
+		env  map[string]string // environment variables
+		opts []Option          // transport spec options
+
+		empty bool // whether this test case is empty
+
+		wantErrors    []string
+		wantOutbounds []string
+	}
+
+	inboundTests := []inboundTest{
+		{desc: "no inbound", empty: true},
+		{
+			desc:          "simple inbound",
+			cfg:           attrs{"tchannel": attrs{"address": ":4040"}},
+			wantTransport: &wantTransport{Address: ":4040"},
+		},
+		{
+			desc:          "inbound interpolation",
+			cfg:           attrs{"tchannel": attrs{"address": ":${PORT}"}},
+			env:           map[string]string{"PORT": "4041"},
+			wantTransport: &wantTransport{Address: ":4041"},
+		},
+		{
+			desc:       "empty address",
+			cfg:        attrs{"tchannel": attrs{"address": ""}},
+			wantErrors: []string{"inbound address is required"},
+		},
+		{
+			desc:       "missing address",
+			cfg:        attrs{"tchannel": attrs{}},
+			wantErrors: []string{"inbound address is required"},
+		},
+		{
+			desc: "too many inbounds",
+			cfg: attrs{
+				"tchannel":  attrs{"address": ":4040"},
+				"tchannel2": attrs{"address": ":4041", "type": "tchannel"},
+			},
+			wantErrors: []string{"at most one TChannel inbound may be specified"},
+		},
+		{
+			desc:       "WithChannel fails",
+			cfg:        attrs{"tchannel": attrs{"address": ":4040"}},
+			opts:       []Option{WithChannel(someChannel)},
+			wantErrors: []string{"TransportSpec does not accept WithChannel"},
+		},
+		{
+			desc:       "ServiceName fails",
+			cfg:        attrs{"tchannel": attrs{"address": ":4040"}},
+			opts:       []Option{ServiceName("zzzzzzzzz")},
+			wantErrors: []string{"TransportSpec does not accept ServiceName"},
+		},
+		{
+			desc:       "ListenAddr fails",
+			cfg:        attrs{"tchannel": attrs{"address": ":4040"}},
+			opts:       []Option{ListenAddr(":8080")},
+			wantErrors: []string{"TransportSpec does not accept ListenAddr"},
+		},
+	}
+
+	outboundTests := []outboundTest{
+		{desc: "no outbound", empty: true},
+		{
+			desc: "simple outbound",
+			cfg: attrs{
+				"myservice": attrs{
+					"tchannel": attrs{
+						"peer": "127.0.0.1:4040",
+					},
+				},
+			},
+		},
+		{
+			desc: "outbound interpolation",
+			env:  map[string]string{"SERVICE_PORT": "4040"},
+			cfg: attrs{
+				"myservice": attrs{
+					"tchannel": attrs{
+						"peer": "127.0.0.1:${SERVICE_PORT}",
+					},
+				},
+			},
+		},
+		{
+			desc: "outbound bad peer list",
+			cfg: attrs{
+				"myservice": attrs{
+					"tchannel": attrs{"least-pending": "wat"},
+				},
+			},
+			wantErrors: []string{
+				`failed to configure unary outbound for "myservice"`,
+				`failed to read attribute "least-pending": wat`,
+			},
+		},
+	}
+
+	runTest := func(t *testing.T, inbound inboundTest, outbound outboundTest) {
+		env := make(map[string]string)
+		for k, v := range inbound.env {
+			env[k] = v
+		}
+		for k, v := range outbound.env {
+			env[k] = v
+		}
```

Comment:
Should we fail the test if the inbound and outbound environments stomp on each other?